src/dmd/backend/cod4.d

```diff
@@ -1745,17 +1733,19 @@ void cddivass(ref CodeBuilder cdb,elem *e,regm_t *pretregs)
         return;
     }
 
+    code cs = void;
+
+    //printf("cddivass(e=%p, *pretregs = %s)\n",e,regm_str(*pretregs));
+    char uns = tyuns(tyml) || tyuns(e2.Ety);
+    uint sz = _tysize[tyml];
+
+    uint rex = (I64 && sz == 8) ? REX_W : 0;
+    uint grex = rex << 16;          // 64 bit operands
+
     if (sz <= REGSIZE)                  // if word or byte
     {
-        isbyte = (sz == 1);               // 1 for byte operation
-        resreg = AX;                    // result register for * or /
-        if (uns)                        // if uint operation
-            opr = 6;                    // DIV
-        else                            // else signed
-            opr = 7;                    // IDIV
-        if (op == OPmodass)
-            resreg = DX;                // remainder is in DX
-
+        uint isbyte = (sz == 1);        // 1 for byte operation
```

Comment:
you can use it directly as assert condition as it's only used there

Reply:
It's also used further down. I left it there because I'm not sure how to do byte operations better.

Reply:
ok I've missed that.